### PR TITLE
docs: clarify that app config is not available (yet) in nitro

### PR DESCRIPTION
**Pull request by ijkml**
### 🔗 Linked issue

Discussion:  #19180

### ❓ Type of change

- [x] 📖 Documentation (updates to the documentation or readme)
- [ ] 🐞 Bug fix (a non-breaking change that fixes an issue)
- [ ] 👌 Enhancement (improving an existing functionality like performance)
- [ ] ✨ New feature (a non-breaking change that adds functionality)
- [ ] ⚠️ Breaking change (fix or feature that would cause existing functionality to change)

### 📚 Description

Provide clarity on use of `useAppConfig()` in `server/api/`.

### 📝 Checklist

- [x] I have linked an issue or discussion.
- [x] I have updated the documentation accordingly.

### CODE REVIEW

File: docs/2.guide/2.directory-structure/3.app-config.md [hunk 1]
```diff
@@ -37,6 +37,10 @@ export default defineAppConfig({
 
 When adding `theme` to the `app.config`, Nuxt uses Vite or webpack to bundle the code. We can universally access `theme` in both server and browser using [useAppConfig](/docs/api/composables/use-app-config) composable.
```
Comment (Atinux) [inline docs/2.guide/2.directory-structure/3.app-config.md:38]:
```suggestion
When adding `theme` to the `app.config`, Nuxt uses Vite or webpack to bundle the code. We can universally access `theme` in both server-side rendering and client-side using [useAppConfig](/docs/api/composables/use-app-config) composable.
```

File: docs/2.guide/2.directory-structure/3.app-config.md [hunk 2]
```diff
@@ -35,7 +35,11 @@ export default defineAppConfig({
 })
 ```
 
-When adding `theme` to the `app.config`, Nuxt uses Vite or webpack to bundle the code. We can universally access `theme` in both server and browser using [useAppConfig](/docs/api/composables/use-app-config) composable.
+When adding `theme` to the `app.config`, Nuxt uses Vite or webpack to bundle the code. We can universally access `theme` in both server-side rendering and client-side using [useAppConfig](/docs/api/composables/use-app-config) composable.
```
Comment (danielroe) [inline docs/2.guide/2.directory-structure/3.app-config.md:38]: I feel like I'm nitpicking now 😆 

```suggestion
When adding `theme` to the `app.config`, Nuxt uses Vite or webpack to bundle the code. We can universally access `theme` both when server-rendering the page and in the browser using [useAppConfig](/docs/api/composables/use-app-config) composable.
```

Comment (danielroe) [reply docs/2.guide/2.directory-structure/3.app-config.md:38]: Do you think we still need the alert now?

Comment (ijkml) [reply docs/2.guide/2.directory-structure/3.app-config.md:38]: The original text now clearly explains where the composable can be used. So, could totally remove the alert or make it talk about Nitro support coming soon (+10% clarity 😅).

Comment (danielroe) [reply docs/2.guide/2.directory-structure/3.app-config.md:38]: Great! Let's convert it to focus on Nitro support coming soon - we could even link to this issue instead: https://github.com/nuxt/nuxt/issues/14670

**Review by danielroe — APPROVED** [review]

**Review by TheAlexLichter — APPROVED** [review]

**Pull request merged**